how-to/how-to-import-kata-logs-with-fluentd.md

```diff
@@ -0,0 +1,604 @@
+journal, and further managed to then parse out the `logfmt` message into JSON to allow further analysis
+inside Elastic/Kibana.
+
+## Directly importing JSON
+
+The underlying basic data format used by fluentd and Elastic is JSON. If we output JSON
+directly from Kata, that should make overall import and processing of the log entries more efficient.
+
+There are potentially two things we can do with Kata here:
+
+- Get Kata to [output its logs in `JSON` format](https://github.com/kata-containers/runtime#logging) rather
+  than `logfmt`.
+- Get Kata to log directly into a file, rather than via the system journal. This would allow us to not need
+  to parse the systemd format files, and capture the Kata log lines directly. It would also avoid `fluentd`
+  having to potentially parse or skip over many non-Kata related systemd journal that it is not at all
+  interested in.
+
+In theory we could get Kata to post its messages in JSON format to the systemd journal by adding the
+`--log-format=json` option to the Kata runtime, and then swapping the `logfmt` parser for the `json`
+parser, but we would still need to parse the systemd files. We will skip this setup in this document, and
+go directly to a full Kata specific JSON format logfile test.
+
+### JSON in files
+
+Kata runtime has the ability to generate JSON logs directly, rather than its default `logfmt` format. Passing
+the `--log-format=json` argument to the Kata runtime enables this. The easiest way to pass in this extra
+parameter from a [Kata deploy](https://github.com/kata-containers/packaging/tree/master/kata-deploy) installation
+is to edit the `/opt/kata/bin/kata-qemu` shell script (generated by the
+[Kata packaging release scripts](https://github.com/kata-containers/packaging/blob/master/release/kata-deploy-binaries.sh)).
+
+At the same time, we will add the `--log=/var/log/kata-runtime.log` argument to store the kata logs in their
+own file (rather than into the system journal).
+
+```bash
+#!/bin/bash
+/opt/kata/bin/kata-runtime --kata-config "/opt/kata/share/defaults/kata-containers/configuration-qemu.toml" --log-format=json --log=/var/log/kata-runtime.log $@
+```
+
+And then we'll add the fluentd config section to parse that file. Note, we inform the parser that Kata is
+generating timestamps in `iso8601` format. Kata places these timestamps into a field called `time`, which
+is the default field the fluentd parser looks for:
+
+```
+    <source>
+      type tail
+      tag kata-containers
+      path /var/log/kata-runtime.log
+      pos_file /var/log/kata-runtime.pos
+      format json
+      time_format %iso8601
+      read_from_head true
+    </source>
+```
+
+This imports the kata-runtime logs, with the resulting records looking like:
+
+![Kata tags in EFK](./images/efk_direct_from_json.png)
+
+Something to note here is that we seem to have gained an awful lot of identical-ish looking fields in the
```

Comment:
Can you add an actual "Note" - https://github.com/kata-containers/documentation/blob/master/Documentation-Requirements.md#notes.

Reply:
Ack